Problem: lexicographical sorting for signed integers doesn't always work

Matt8898 opened the pull request:
Solution: Flip the first bit of sized INTs and encode unsized INTs as two's complement.

yrashk commented:
Awesome! I think it would make sense to include the sized-comparison-instructions reversal commit into this PR to keep the whole change together. Or do you have a better idea how to organize it?

Matt8898 commented:
I think that's good, can i run cargo fmt?

yrashk commented:
yeah, cargofmt is fine as long as it doesn't touch other parts of the code (the ones that don't affect the actual change)

Matt8898 commented:
I should remove normal INT comparison too right?

yrashk commented:
> I should remove normal INT comparison too right?

What do you mean by normal? unsized? If unsized, then I wouldn't say so, as
unsized integer comparison takes differently sized integers and processes them
as integers, not just lexicographically sortable binaries:

```
PumpkinDB> 0x64 0x0064 UINT/EQUAL?.
0x01
PumpkinDB> 0x64 0x0064 EQUAL?.
0x00
```

Matt8898 commented:
Ok.

yrashk commented:
BTW, while we're at it, b4661ef1481f0a7c304c57f7df6935689cd9ee6c is missing the `Solution` part, it seems.

Matt8898 committed:
Problem: lexicographical sorting for signed integers doesn't always work
Solution: Flip the first bit of sized INTs and encode unsized INTs as two's complement.

yrashk commented:
Also, adding doctests based on the example from #267 would make sense. But we can add those later. If you want, I can handle it after the merge.

Matt8898 commented:
Ok, everything should be fine now.

Matt8898 committed:
Problem: sized integer comparison duplicates lexicographical comparison
